src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -83,14 +101,128 @@ public async Task<IActionResult> Index()
+            }
+
+            switch (options.Filter)
+            {
+                case TenantsFilter.Disabled:
+                    entries = entries.Where(t => t.ShellSettings.State == TenantState.Disabled).ToList();
+                    break;
+                case TenantsFilter.Running:
+                    entries = entries.Where(t => t.ShellSettings.State == TenantState.Running).ToList();
+                    break;
+                case TenantsFilter.Uninitialized:
+                    entries = entries.Where(t => t.ShellSettings.State == TenantState.Uninitialized).ToList();
+                    break;
+            }
+
+            switch (options.OrderBy)
+            {
+                case TenantsOrder.Name:
+                    entries = entries.OrderBy(t => t.Name).ToList();
+                    break;
+                case TenantsOrder.State:
+                    entries = entries.OrderBy(t => t.ShellSettings?.State).ToList();
+                    break;
+                default:
+                    entries = entries.OrderByDescending(t => t.Name).ToList();
+                    break;
+            }
+            var count = entries.Count();
+
+            var results = entries
+                .Skip(pager.GetStartIndex())
+                .Take(pager.PageSize).ToList();
+
+            // Maintain previous route data when generating page links
+            var routeData = new RouteData();
+            routeData.Values.Add("Options.Filter", options.Filter);
+            routeData.Values.Add("Options.Search", options.Search);
+            routeData.Values.Add("Options.SqlProvider", options.SqlProvider);
+            routeData.Values.Add("Options.OrderBy", options.OrderBy);
+
+            var pagerShape = (await New.Pager(pager)).TotalItemCount(count).RouteData(routeData);
+
+            var model = new AdminIndexViewModel
+            {
+                ShellSettingsEntries = results,
+                Options = options,
+                Pager = pagerShape
             };
 
             return View(model);
         }
 
+        [HttpPost]
+        public async Task<IActionResult> Index(BulkActionViewModel model)
+        {
+            foreach(var tenantName in model.TenantNames)
+            {
+                var shellContext = (await GetShellsAsync())
+                    .Where(x => string.Equals(x.Settings.Name, tenantName, StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault();
+
+                if (shellContext == null)
+                {
+                    break;
+                }
+
+                var shellSettings = shellContext.Settings;
+
+                switch (model.BulkAction.ToString())
+                {
+                    case "Disable":
+                        if (string.Equals(shellSettings.Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase))
```

Comment:
This test should not be in the loop, or it should not break the loop.

Reply:
OK. Fixed